### DISCUSSION TITLE
[Question][Hw5] Abstract Semantics

### DISCUSSION THREAD

**Discussion by john-mai-2605** [prosyslab-classroom/cs348-information-security#616] [Q&A]
I dont understand what we have to do for "transfer_atomic, transfer_cond, transfer_phi"
Can anyone give me some hints ...

**Comment by RiceBiscuits** [comment]
`transfer_*` functions calculate a memory state (Abstract memory) that represent after executing given node.

**Comment by john-mai-2605** [comment]
How does transfer_cond differ from filter?

**Comment by RiceBiscuits** [comment]
Filter will refine a variable using `pred`, and `transfer_cond` deals a conditional branches using `Filter`.
And `transfer_cond` should do almost everything except `Filter` can do.
Such as, determine what kind of instructions has been provided from the caller function.

**Comment by john-mai-2605** [comment]
Sorry for too many questions. I have 1 last question.
In transfer phi, I saw that there are iterations in both i and j in the doc. But in the code, there is only 1 list of instr. How should I understand this function? Thank you

**Comment by KAIST-JongchanPark** [comment]
I think #155 might be helpful for @john-mai-2605 's last question

**Comment by john-mai-2605** [comment]
Closed
